app/src/org/commcare/activities/HomeScreenBaseActivity.java

```diff
@@ -917,6 +917,7 @@ private boolean processReturnFromFormEntry(int resultCode, Intent intent) {
             // The form is either ready for processing, or not, depending on how it was saved
             if (complete) {
                 startUnsentFormsTask(false, false);
+                CommCareApplication.instance().scheduleEntityCacheInvalidation();
```

Comment:
Two quick things:

1- If the form is 'complete' and not 'unsent' that means it wasn't processed, right? A little concerned about the possibility that the UnsentForms task might not have finished processing the form before the actual underlying work here is performed. 

2 - Seems a little odd to queue this if there is no caching going on with the current app. I still don't think we're ready to view caching as the default, so I'm interested  (where it's not a ton of work) in still minimizing the new surface area non-cache-and-index apps are exposed to from the changes.

Reply:
1. https://github.com/dimagi/commcare-android/pull/2955/commits/867c25462ac0264a267a21b97cd007ee864a3347
2. https://github.com/dimagi/commcare-android/pull/2955/commits/be429057100fee52a003f241385a4ed77d8e2d2c